contracts/bprotocol/avatar/AbsCToken.sol

```diff
@@ -83,9 +83,15 @@ contract AbsCToken is Cushion {
         returns (uint256)
     {
         uint256 amtToRepayOnCompound = _untopPartial(cToken, repayAmount);
-        if(amtToRepayOnCompound > 0) return cToken.repayBorrow(amtToRepayOnCompound); // in case of err, tx fails at BToken
-        _score().updateDebtScore(address(this), address(cToken), -toInt256(repayAmount));
-        return 0; // no-err
+        uint256 result = 0;
+        if(amtToRepayOnCompound > 0) {
+            IERC20 underlying = cToken.underlying();
+            underlying.safeApprove(address(cToken), 0);
```

Comment:
* ideally we should approve infinity at certain point
* ** there is something annoying about COMP token, where you cannot approve uint(-1) but only uint64(-1).
* *** technically after repay the allowance should be 0, so maybe no need to reset it.

Reply:
need an infinite allowance. Check for COMP token when full allownace.

Reply:
can we remove `underlying.safeApprove(address(cToken), 0);`

Reply:
give exact allowance (without reset)
have a backdoor function to reset an allowance (per token and dst)
same apporach in `mint`.
Check if applicable to redeem.